src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -284,6 +287,52 @@ public override void VisitModuleDeclarationSyntax(ModuleDeclarationSyntax syntax
                     return singleDeclaredType;
                 }
 
+                // We need to validate all of the parameters and outputs to make sure they are valid types.
+                // This is where we surface errors for 'unknown' resource types.
+                if (singleDeclaredType is ModuleType moduleType &&
+                    moduleType.Body is ObjectType objectType)
```

Comment:
>Body [](http://example.com/codeflow?start=31&length=4)

You may need to handle loops and/or conditionals here. There's a method that can get the body for all those cases (`TryGetBody` or something like that).

Reply:
Are you sure? This is looking at types not syntax. L281 ^^^ handles the array case.